### PR TITLE
Some events refactoring

### PR DESCRIPTION
**Pull request by anton-ryzhov**
I've done some work about events mechanics.
1. Removed legacy and useless "roster_update".
2. Duplicated some similar events
3. Updated events reference document.

Apply it partially or entirely if you'll find it useful.

### CODE REVIEW

File: sleekxmpp/features/feature_mechanisms/mechanisms.py [hunk 1]
```diff
@@ -234,6 +235,6 @@ def _handle_fail(self, stanza):
         """SASL authentication failed. Disconnect and shutdown."""
         self.attempted_mechs.add(self.mech.name)
         log.info("Authentication failed: %s", stanza['condition'])
-        self.xmpp.event("failed_auth", stanza, direct=True)
+        self.xmpp.event("failed_auth", direct=True)
```
Comment (legastero) [inline sleekxmpp/features/feature_mechanisms/mechanisms.py]: No, the failed stanza can include additional error messages, so it must be included in the event.

**Comment by legastero** [comment]
Except for the one comment I added, this looks great. I'll merge it in what that is fixed.

File: sleekxmpp/features/feature_mechanisms/mechanisms.py (hunk 1 above)
Comment (anton-ryzhov) [inline sleekxmpp/features/feature_mechanisms/mechanisms.py]: Legacy auth also fires this event and don't append stanza. I'll fix it.

**Comment by anton-ryzhov** [comment]
I've returned that stanza in `failed_auth`.

**Pull request merged**